web/v2/player/index.ts

```diff
@@ -227,3 +216,31 @@ export interface APIPlayer {
    */
   vtcHistory: APIPlayerCompanyHistory[] | null;
 }
+
+/** Properties specific to the player resource when VTC information is hidden */
+interface _VTCHistoryHidden extends Pick<_Base, 'displayVTCHistory' | 'vtcHistory'> {
+  displayVTCHistory: false;
+  vtcHistory: null;
+}
+
+/** Properties specific to the player resource when VTC information is public */
+interface _VTCHistoryPublic extends Pick<_Base, 'displayVTCHistory' | 'vtcHistory'> {
+  displayVTCHistory: true;
+  vtcHistory: APIPlayerCompanyHistory[];
+}
+type _VTCHistory = _VTCHistoryHidden | _VTCHistoryPublic;
+
+/** Properties specific to the player resource when ban information is hidden */
+interface _BansHidden extends Pick<_Base, 'displayBans' | 'bansCount'> {
+  displayBans: false;
+  bansCount: null;
+}
+
+/** Properties specific to the player resource when ban information is public */
+interface _BansPublic extends Pick<_Base, 'displayBans' | 'bansCount'> {
+  displayBans: true;
+  bansCount: number;
+}
+type _BanData = _BansHidden | _BansPublic;
```

Comment:
I get why the union is used, although, cannot we use the same approach as you used for the traffic API, i.e., without these "temporary" interfaces? I feel like they only make the code messier and could possibly leak to the build. Furthermore, the attributes lack documentation. Lastly, if we decide to keep them, it would have been nice to make the interface documentation consistent with the rest of the code base.

Reply:
Simplified the initial 3-step types into 2-steps and named them more specifically. As for property documentation, since all the types inherit from `Pick<Base, ...>` the documentation also carries over (at least in VSCode, but I assume it isn't much different in other editors).

Reply:
The 2-step types are indeed better; thank you for the change. And the documentation indeed carries over. I did not notice that they are commented earlier in the file already. :)